she_codes_news/news/middleware.py

```diff
@@ -0,0 +1,21 @@
+from urllib import parse
+from zoneinfo import ZoneInfo
+
+from django.conf import settings
+from django.utils import timezone
+from django.utils.deprecation import MiddlewareMixin
+
+
+class TimezoneMiddleware(MiddlewareMixin):
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        # get timezone name from browser cookie
+        tz_name = request.COOKIES.get('timezone') or settings.TIME_ZONE
```

Comment:
will this always be true, since `settings.TIME_ZONE` is set?

Reply:
yeah I'm just wondering whether there needs to be an if statement then?

Reply:
I'd also suggest 

```
request.COOKIES.get('timezone', settings.TIME_ZONE)
```

Reply:
oh yes!! it totally could be